### PR TITLE
planner: unify the log format

### PR DESCRIPTION
**Pull request by XuHuaiyu**
### What problem does this PR solve? 
unify the log format in package planner

### What is changed and how it works?
Using 

### Check List 

Tests 

N/A

Code changes

unify log format

Side effects

N/A

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: planner/core/logical_plans.go [hunk 1]
```diff
@@ -497,7 +499,7 @@ func (ds *DataSource) deriveIndexPathStats(path *accessPath) (bool, error) {
 	if path.indexFilters != nil {
 		selectivity, _, err := ds.stats.HistColl.Selectivity(ds.ctx, path.indexFilters)
 		if err != nil {
-			log.Warnf("An error happened: %v, we have to use the default selectivity", err.Error())
+			logutil.Logger(context.Background()).Warn("an error happened, use the default selectivity", zap.Error(err))
```
Comment (zz-jason) [inline planner/core/logical_plans.go]: how about:
```suggestion
			logutil.Logger(context.Background()).Warn("calculate selectivity faild, use selection factor", zap.Error(err), zap.Float64("selection factor", selectionFactor))
```

File: planner/core/rule_join_elimination.go [hunk 2]
```diff
@@ -76,7 +79,7 @@ func (o *outerJoinEliminator) isAggColsAllFromOuterTable(outerPlan LogicalPlan,
 	for _, col := range aggCols {
 		columnName := &ast.ColumnName{Schema: col.DBName, Table: col.TblName, Name: col.ColName}
 		if c, err := outerPlan.Schema().FindColumn(columnName); c == nil {
-			log.Warn(err)
+			logutil.Logger(context.Background()).Warn("outerJoinEliminator raises an error", zap.Error(err))
```
Comment (zz-jason) [inline planner/core/rule_join_elimination.go]: there are three places which logs `outerJoinEliminator raises an error`. Could you also print the step information during the outer join elimination to help identify the problems?

File: planner/core/rule_join_elimination.go [hunk 3]
```diff
@@ -91,7 +94,7 @@ func (o *outerJoinEliminator) isParentColsAllFromOuterTable(outerPlan LogicalPla
 	for _, col := range parentSchema.Columns {
 		columnName := &ast.ColumnName{Schema: col.DBName, Table: col.TblName, Name: col.ColName}
 		if c, err := outerPlan.Schema().FindColumn(columnName); c == nil {
-			log.Warn(err)
+			logutil.Logger(context.Background()).Warn("outerJoinEliminator raises an error", zap.Error(err))
```
Comment (lamxTyler) [inline planner/core/rule_join_elimination.go]: Why use `context.Background`? Why not just use `log.Warn`?

Comment (XuHuaiyu) [reply planner/core/rule_join_elimination.go]: After the discussion, we use context.Background() now.

File: planner/core/rule_join_elimination.go (hunk 2 above)
Comment (eurekaka) [reply planner/core/rule_join_elimination.go]: These warning logs should have been removed by https://github.com/pingcap/tidb/pull/9631.

File: planner/core/stats.go [hunk 4]
```diff
@@ -194,7 +196,7 @@ func (lt *LogicalTopN) DeriveStats(childStats []*property.StatsInfo) (*property.
 func getCardinality(cols []*expression.Column, schema *expression.Schema, profile *property.StatsInfo) float64 {
 	indices := schema.ColumnsIndices(cols)
 	if indices == nil {
-		log.Errorf("Cannot find column %v indices from schema %s", cols, schema)
+		logutil.Logger(context.Background()).Error("cannot find columns in schema", zap.Any("columns", cols), zap.String("schema", schema.String()))
```
Comment (eurekaka) [inline planner/core/stats.go]:
```suggestion
		logutil.Logger(context.Background()).Error("column not found in schema", zap.Any("columns", cols), zap.String("schema", schema.String()))
```

**Pull request merged**